backend.py

```diff
@@ -312,27 +318,38 @@ def get_robot_paths():
 def create_robots(board):
     """
     Place robots on start tiles.
+
     board: dictionary returned by get_board()
-    Return list of robots on the start tiles of the board.
     Initialize Robot objects on the start tiles coordinates with random
     choice of robot's avatar on particular tile.
     Once the robot is randomly chosen, he is removed from the list
     (he cannot appear twice on the board).
-    On the beginning all the Robot objects have implicit 0 degree direction.
+    Robots are placed on board in the direction of their start tiles.
+    The robots are ordered according to their start tiles.
     """
-    start_coordinates = get_start_coordinates(board)
+    start_tiles = get_start_tiles(board)
     robot_paths = get_robot_paths()
-    robots_start = []
-    for coordinate in start_coordinates:
-        # Condition to assure no exception in case robot_paths is shorter
-        # than coordinate's list.
+    robots_on_start = []
+    for start_tile_number in start_tiles:
+        # Condition to assure no exception in case there is less robot_paths
+        # than tiles.
         if robot_paths:
+            # Get graphics for the robot
             paths = random.choice(robot_paths)
             robot_paths.remove(paths)
             path, path_front = paths
-            robot = Robot(Direction.N, path, path_front, coordinate)
-            robots_start.append(robot)
-    return robots_start
+
+            # Get direction and coordinates for the robot
+            initial_direction = start_tiles[start_tile_number]["tile_direction"]
+            initial_coordinates = start_tiles[start_tile_number]["coordinates"]
+
+            # Create a robot, add him to robot's list
+            robot = Robot(initial_direction, path, path_front, initial_coordinates)
+            robots_on_start.append(robot)
+
+            # Set robot on the index of his starting tile (1 -> 0 and so on)
+            robots_on_start[start_tile_number - 1] = robot
```

Comment:
Tento řádek je zbytečný. Seznam se plní od indexu 0.  Navíc ten komentář nad tím je trochu zavádějící. Podle první části bych řekla, že chceš mít roboty indexu rovném číslu startovního políčka, takže něco jako `robots_on_start[1] ->  [1: (1, 1), Direction.N]`. Ale příklad v závorce i zápis pod tím vede k `robots_on_start[0] ->  [1: (1, 1), Direction.N]`. Jak bys ten seznam chtěla mít řazený i třeba vzhledem k testovacímu frameworku?

Reply:
Chci aby roboti v seznamu odpovidali poradi startovacich policek [robot1, robot2, robot3, robot4]. 
Klice ve slovniku jsou hodnoty tohoto poradi, ale slovnik nemusi byt vnitrne sezaren podle tech klicu, a proto iterace po klicich a pridavani hodnot do seznamu nemusi ve vsech pythonech fungovat stejne (3.5 dela potize). 
Ted me napadl jistejsi zpusob, jak to udelat: iterace po rozsahu delky slovniku, kde budu plnit seznam robotu podle vypocitaneho klice. (`for i in range(len(slovnik)): seznam.append(slovnik[i+1]`). Pak by meli byt roboti skutecne ok.

Reply:
Můžeš použít [seřazený slovník](https://docs.python.org/3/library/collections.html#collections.OrderedDict) z `collections`, ten je i v Pythonu 3.5. (Od 3.6 jsou takhle seřazené i normální slovníky, ale OrderedDict má pořád pár metod navíc).

Reply:
Diky za odkaz, povedlo se naimplementovat :)